src/common/tusb_fifo.c

```diff
@@ -30,6 +30,10 @@
 #include "osal/osal.h"
 #include "tusb_fifo.h"
 
+#if defined(__ICCARM__)
+#pragma diag_suppress = Pa082
```

Comment:
what is the Pa082 that we need to suppress with IAR ? I would prefer to fix the warning with the code rather than suppress it with pragma

Reply:
Warning is `Warning[Pa082]: undefined behavior: the order of volatile accesses is undefined in this statement`
Because both `wr_idx` and `rd_idx` are volatile, IAR don't know which one should be accessed first.

In order to fix it we need to add temporary variables:
```
uint16_t tu_fifo_count(tu_fifo_t* f)
{
    uint16_t wr_idx = f->wr_idx;
  return _tu_fifo_count(f, wr_idx, f->rd_idx);
}
```

Reply:
CC @PanRe

Reply:
@HiFiPhile ah, I know this warning, have been encountered it before. IAR is pretty smart  about this. There is quite a lot of wr and rd idx passing to a function at same time. So yes, it makes sense to suppress the warning as you did here. We may come back later to "fix" it whenever having time. Please also put the warning message of  `Warning[Pa082]: undefined behavior: the order of volatile accesses is undefined in this statement` so that we could know why we suppress the warning.